extensions/amp-app-banner/1.0/component/android.js

```diff
@@ -0,0 +1,106 @@
+import {WindowInterface} from '#core/window/interface';
+
+import {docInfoService} from '#preact/services/document';
+import {platformService} from '#preact/services/platform';
+import {timerService} from '#preact/services/timer';
+import {urlService} from '#preact/services/url';
+import {viewerService} from '#preact/services/viewer';
+import {xhrService} from '#preact/services/xhr';
```

Comment:
# The majority of feedback on this PR is focused on these `#preact/services`, so I'd love to get feedback on how to improve this approach.

Without knowing the full context of how Bento components should differ from Amp Classic components, I started by porting (or stubbing) most of these services, so that the `app-banner` component (and future Bento components) can be easily ported to Bento.  That's why many of these methods are unnecessarily marked `async` and are often just thin wrappers around storage / fetch, etc.  This is not "good code", but it allowed me to quickly and accurately port the `app-banner` logic.

Realistically, out of all these 7 "services", there's about 4 functions that are actually useful.  It would be very easy to rewrite these parts as simple utilities, instead of trying to keep a "service-oriented" approach.  

It would be nice, if these Bento-specific utilities were still reusable, so I'd prefer to keep them in a common location, like `#preact/utils/*` ... so ... does this sound like it would be a welcome improvement?  Any other suggestions?

Reply:
Follow up: these changes have been pushed.  

The `preact/utils` folder is now a place for putting common Bento utils, so that we don't end up repeating things across every Bento component.

The `preact/services` have been removed, and only a few needed parts have been moved to `utils`.